### PR TITLE
Core 630 join team checkout

### PR DESCRIPTION
**Pull request by roger-in-kiva**

### CODE REVIEW

**Comment by mcstover** [comment]
@michelleinez We have fallback mechanisms in our actual `/checkout` page to handle promo users that exit the single page experience and end up going through the standard checkout. We've already added support for handling checkout properly and the employee verification form as the components are shared and we just needed to include the logic to activate them.

This work also adds in the team join prompt.

I've added you to this PR b/c we'll need to do some deliberate testing on this code to ensure it's functioning properly for promo account AND that the standard checkout flow isn't obstructed. If you are both ready to roll on that testing we can merge  and start, otherwise I'd suggest waiting to merge until after lunch on Monday so we can ready the primary release without this included.

File: src/pages/Checkout/CheckoutPage.vue [hunk 1]
```diff
@@ -212,6 +212,13 @@
 				@credit-removed="handleCreditRemoved"
 				@promo-opt-out-lightbox-closed="handleCancelPromoOptOut"
 			/>
+			<campaign-join-team-form
+				v-if="this.showTeamForm"
+				:campaign-name="campaignPartnerName"
+				:team-id="this.teamId"
+				:promo-id="this.promoFundId"
+				@team-process-complete="handleTeamJoinProcess"
```
Comment (mcstover) [inline src/pages/Checkout/CheckoutPage.vue:220]: @roger-in-kiva Need to remove `this` on all of these fields.

Comment (mcstover) [reply src/pages/Checkout/CheckoutPage.vue:220]: I see that's how it's setup in the CCLandingPage too. But `this` shouldn't be necessary...

**Review by mcstover — APPROVED** [review]

File: src/pages/Checkout/CheckoutPage.vue (hunk 1 above)
Comment (roger-in-kiva) [reply src/pages/Checkout/CheckoutPage.vue:221]: Agreed!

**Pull request merged**